lib/apytypes/_array_functions.py

```diff
@@ -834,55 +900,57 @@ def arange(
 # Helpers
 # =============================================================================
-def _determine_array_type(int_bits, frac_bits, exp_bits, mantissa_bits):
+def _determine_array_type(int_bits, frac_bits, bits, exp_bits, mantissa_bits):
     from apytypes import APyFixedArray, APyFloatArray
 
-    if int_bits is not None and frac_bits is not None:
-        return APyFixedArray
-    elif exp_bits is not None and mantissa_bits is not None:
-        return APyFloatArray
-    raise ValueError("You need to specify wordlengths in this function")
+    if exp_bits is None and mantissa_bits is None:
+        if int_bits is not None and frac_bits is not None:
+            return APyFixedArray
+        elif bits is not None and int_bits is not None:
+            return APyFixedArray
+        elif bits is not None and frac_bits is not None:
+            return APyFixedArray
+    elif bits is None and int_bits is None and frac_bits is None:
+        if exp_bits is not None and mantissa_bits is not None:
+            return APyFloatArray
+
+    raise ValueError("Could not determine array type from bit-specifiers")
 
 
-def _normalize_fill_value(
+def _determine_fill_value(
     fill_value,
     int_bits=None,
     frac_bits=None,
+    bits=None,
     exp_bits=None,
     mantissa_bits=None,
 ):
     from apytypes import APyFixed, APyFloat
 
     if isinstance(fill_value, (int, float)):
-        if int_bits is not None and frac_bits is not None:
+        if frac_bits is None and exp_bits is None:
             return APyFixed.from_float(
-                fill_value, int_bits=int_bits, frac_bits=frac_bits
+                fill_value, int_bits=int_bits, frac_bits=frac_bits, bits=bits
             )
-        elif exp_bits is not None and mantissa_bits is not None:
+        else:
             return APyFloat.from_float(
                 fill_value, exp_bits=exp_bits, man_bits=mantissa_bits
             )
-        else:
-            raise ValueError(
-                "You need to specify wordlengths if input is a Python float or integer"
-            )
-
     elif isinstance(fill_value, APyFixed):
         int_bits = int_bits if int_bits is not None else fill_value.int_bits
         frac_bits = frac_bits if frac_bits is not None else fill_value.frac_bits
+        bits = bits if bits is not None else fill_value.bits
         return fill_value.cast(int_bits, frac_bits)
-
     elif isinstance(fill_value, APyFloat):
         exp_bits = exp_bits if exp_bits is not None else fill_value.exp_bits
         mantissa_bits = (
             mantissa_bits if mantissa_bits is not None else fill_value.man_bits
         )
         return fill_value.cast(exp_bits, mantissa_bits, fill_value.bias)
 
-    else:
-        raise ValueError(
-            "Only int, float, APyFloat and APyFixed are supported for fill_value"
-        )
+    raise ValueError(
+        "Only int, float, APyFloat and APyFixed are supported for fill_value"
+    )
```

Comment:
In some code above, you did the exact opposite: added an else-clause at the end. I do not have a strong opinon which is preferred, but seems a it inconsistent to do both types of modifications.